heads.go

```diff
@@ -56,11 +56,7 @@ func (hh *heads) write(store ds.Write, c cid.Cid, height uint64) error {
 }
 
 func (hh *heads) delete(store ds.Write, c cid.Cid) error {
-	err := store.Delete(hh.key(c))
-	if err == ds.ErrNotFound {
-		return nil
-	}
-	return err
```

Comment:
I'm not sure we should change this. This changes internal behaviour which was explicit in detecting that we are trying to remove an unexistent head. I don't know if this check is warranted or not (without diving deeper), but it is not related to this PR and I don't want to risk a subtle bug being introduced.

Reply:
Sure, I can remove this part.